src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -106,7 +106,6 @@ public async Task GetClusterWideInfoPackage()
             using (ctx.OpenReadTransaction())
                 topology = ServerStore.GetClusterTopology(ctx);
 
-            var timeoutInSecPerNode = GetIntValueQueryString("timeoutInSecPerNode", false) ?? 60;
```

Comment:
We need the debug package to produce a result, this depends on the number of cores that the machine has and the number of databases in it. It's impossible to know how much time we'll need.

Note that the operation can still be gracefully canceled and it will collect all the relevant data until then.

Reply:
By saying "gracefully canceled" you're referring to request aborted, right?

https://github.com/ravendb/ravendb/blob/f6cffb7819098b0e7717a03b7b44a9ddb0c7e6e2/src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs#L110

https://github.com/ravendb/ravendb/blob/f6cffb7819098b0e7717a03b7b44a9ddb0c7e6e2/src/Raven.Server/Web/RequestHandler.cs#L740-L743

Reply:
IMO you need to introduce a timeout here, even if it is unreasonably high like 30min. I would like to avoid having 12h requests hanging here. If this takes more than e.g. 30min for sure we have different issues.

Reply:
@arekpalinski, the operation can be canceled (this is a server wide operation) by calling kill operation:
https://github.com/ravendb/ravendb/blob/f6cffb7819098b0e7717a03b7b44a9ddb0c7e6e2/src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs#L140

Reply:
I increased to timeout to 1 hour.